mesteriApplication/src/main/java/com/intern/alexx/repository/impl/MesterRepositoryImp.java

```diff
@@ -144,105 +127,137 @@ public Mester getById(Mester mester) {
+		return mester;
+	}
 
-		String sql1 = generateSql.createQueryForCountElements(searchCriteria);
-		String sql2 = generateSql.createQueryForElements(searchCriteria);
-		MyPage page = null;
-		Mester mester=null;
+	public void addMesterIntoDB(Mester mester, PreparedStatement ps) throws SQLException {
+
+		ps.setString(1, mester.getFirstName());
+		ps.setString(2, mester.getLastName());
+		ps.setString(3, mester.getDescription());
+		ps.setString(4, mester.getLocation());
+	}
+
+	public int executeSqlCountStatement(MesterSearchCriteria searchCriteria) {
 		Connection conn = null;
+		String sql = generateSql.createQueryForCountElements(searchCriteria);
+		int totalElements = 0;
 		try {
 			conn = dataSource.getConnection();
-			PreparedStatement ps = conn.prepareStatement(sql1);
-			PreparedStatement ps2 = conn.prepareStatement(sql2);
-		
-			int nrParm=1;
-				if (searchCriteria.getFirstName() != null) {
-					 ps.setString(nrParm, searchCriteria.getFirstName());
-					 ps2.setString(nrParm, searchCriteria.getFirstName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getLastName() != null) {
-					 ps.setString(nrParm, searchCriteria.getLastName());
-					 ps2.setString(nrParm, searchCriteria.getLastName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getLocation() != null) {
-					 ps.setString(nrParm, searchCriteria.getLocation());
-					 ps2.setString(nrParm, searchCriteria.getLocation());
-					 nrParm++;
-					 }
-				if (searchCriteria.getSpecialityName() != null) {
-					 ps.setString(nrParm, searchCriteria.getSpecialityName());
-					 ps2.setString(nrParm, searchCriteria.getSpecialityName());
-					 nrParm++;
-					 }
-				if (searchCriteria.getEmail() != null) {
-					 ps.setString(nrParm, searchCriteria.getEmail());
-					 ps2.setString(nrParm, searchCriteria.getEmail());
-					 nrParm++;
-					 }
-				if (searchCriteria.getPhoneNumber() != null) {
-					 ps.setString(nrParm, searchCriteria.getPhoneNumber());
-					 ps2.setString(nrParm, searchCriteria.getPhoneNumber());
-					 nrParm++;
-					 }
-				if (searchCriteria.getRating() != null) {
-					 ps.setInt(nrParm, searchCriteria.getRating());
-					 ps2.setInt(nrParm, searchCriteria.getRating());
-					 nrParm++;
-					  }
-				if (searchCriteria.getPrice() != null) {
-					 ps.setString(nrParm, searchCriteria.getPrice());
-					 ps2.setString(nrParm, searchCriteria.getPrice());
-					 nrParm++;
-					 
-				}
-				
-				if(searchCriteria.getPageNumber() != null) {
-				page.setPageNumber(searchCriteria.getPageNumber());
-				}
-				if(searchCriteria.getPageSize() != null) {
-				page.setPageSize(searchCriteria.getPageSize());
-				}
+
+			PreparedStatement ps = conn.prepareStatement(sql);
+			System.out.println(sql);
+			verifyParam(ps, searchCriteria);
+
 			ResultSet resultSet = ps.executeQuery();
-		
-			page.setTotalRezults(resultSet.getInt(1));
+			if (resultSet.next()) {
+				totalElements = resultSet.getInt("total");
+			}
 			ps.close();
```

Comment:
statement nu se inchide tot timpul ( e.g. daca se arucna o exceptie). Trebuie inchis in finally